lib/routes/Character/components/CharacterDialog/components/blocks/BlockDicePool.tsx

```diff
@@ -77,59 +87,77 @@ export function BlockDicePool(
         >
           <Grid item>
             <Box>
-              <Pool
-                fontSize="1.2rem"
-                borderRadius="8px"
-                selected={isSelected}
-                clickable={canRoll}
-                borderStyle={hasCommands ? "solid" : "dashed"}
-                onClick={() => {
-                  if (!canRoll) {
-                    return;
-                  }
-
-                  props.onPoolClick({
-                    blockId: props.block.id,
-                    blockType: props.block.type,
-                    label: props.block.label,
-                    commandOptionList: commandOptionList,
+              <DiceMenuForCharacterSheet
+                commandGroupIds={commands}
+                onChange={(newCommandIds) => {
+                  props.onMetaChange({
+                    ...props.block.meta,
+                    commands: newCommandIds,
                   });
                 }}
-              >
-                <Grid
-                  container
-                  spacing={1}
-                  alignItems="center"
-                  justify="center"
-                >
-                  {!hasCommands && (
-                    <Grid item>
-                      <FormHelperText
-                        className={css({
-                          margin: "0",
-                          padding: ".5rem",
-                        })}
-                      >
-                        {t("character-dialog.helper-text.empty-dice-pool")}
-                      </FormHelperText>
-                    </Grid>
-                  )}
-                  {blockCommandGroups.map((commandGroup, index) => {
-                    return (
-                      <Grid item key={index}>
-                        <Tooltip title={commandGroup.label}>
-                          <commandGroup.icon
+                render={(diceMenuProps) => (
+                  <Pool
+                    fontSize="1.2rem"
```

Comment:
Using the render props pattern here to have a component that manages the dice menu what is selected and everything but I can decide using the renderProps when its opened or closed